Mathlib/RingTheory/Noetherian.lean

```diff
@@ -178,82 +179,100 @@ end Module
 
 section
 
-variable {R : Type _} {M : Type _} {P : Type _}
+-- Porting note: adding explicit universes
+universe u v w u'
+
+variable {R : Type u} {M : Type v} {P : Type w}
 
 variable [Ring R] [AddCommGroup M] [AddCommGroup P]
 
 variable [Module R M] [Module R P]
 
 open IsNoetherian
 
-include R
-
-theorem isNoetherian_of_ker_bot [IsNoetherian R P] (f : M →ₗ[R] P) (hf : f.ker = ⊥) :
-    IsNoetherian R M :=
-  isNoetherian_of_linearEquiv (LinearEquiv.ofInjective f <| LinearMap.ker_eq_bot.mp hf).symm
+-- Porting note: more typeclass search juggling
```

Comment:
This goes from a perfectly readable statement to a huge `@`-jungle.
I'm quite worried about what this means for the 800 files that depend on this. Do you know if the original statement works when `etaExperiment` is enabled?

Reply:
There are only 2-3 TC searches that occur repeatedly and cause all of this mess. `etaExperiment` fixes those though it did error on one other thing for me. I did some digging into the search belt my last commit but I might be able to pin down the issue more.